test/acceptance/features/bindAppToServiceAnnotationsOptional.feature

```diff
@@ -0,0 +1,173 @@
+@annotations
+@disabled
+@external-feedback
+Feature: Bind an application to a service using optional annotations
+
+    As a user of Service Binding Operator
+    I want to bind application to services that expose optional bindable information
+    via annotations placed on service's CRD or CR.
+
+    Background:
+        Given Namespace [TEST_NAMESPACE] is used
+        * Service Binding Operator is running
+        * OLM Operator "custom_app" is running
+        * CustomResourceDefinition backends.stable.example.com is available
+
+    Scenario: Bind to the application with annotations and with optional source field present
+        Given Generic test application is running
+        And The Custom Resource is present
+            """
+            apiVersion: "stable.example.com/v1"
+            kind: AppConfig
+            metadata:
+                name: $scenario_id
+                annottions:
+                    service.binding/uri: "path={.spec.uri},optional=exclude"
+                    service.binding/image: "path={.spec.image}"
+            spec:
+                uri: "youknow.where"
+                image: "busybox"
+            """
+        When Service Binding is applied
+            """
+            apiVersion: binding.operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: $scenario_id
+            spec:
+                services:
+                  - group: stable.example.com
+                    version: v1
+                    kind: AppConfig
+                    name: $scenario_id
+                application:
+                    group: apps
+                    version: v1
+                    resource: deployments
+                    name: $scenario_id
+            """
+        Then Service Binding is ready
+        * Content of file "/bindings/$scenario_id/uri" in application pod is
+            """
+            youknow.where
+            """
+        * Content of file "/bindings/$scenario_id/image" in application pod is
+            """
+            busybox
+            """
+
+    Scenario: Bind to the application with annotations and with missing optional source field to be excluded
+        Given Generic test application is running
+        And The Custom Resource is present
+            """
+            apiVersion: "stable.example.com/v1"
+            kind: AppConfig
+            metadata:
+                name: $scenario_id
+                annottions:
+                    service.binding/uri: "path={.spec.uri},optional=exclude"
```

Comment:
The value for `optional` should be `true` or `false` as explained in the story.

Reply:
We have deciced on a community call that we make the feature a bit less complicated. The idea is that if the source data is optional, the annotation will have just the `optional` flag and it would exclude the items from the projected binding data if the source data is not available. If `optional` is not set, than the binding collection would fail if the source data is not available.

Reply:
Updated the scenario according to the above.